### PR TITLE
FIX avoid upload files problem + avoid warnings php8 + TODO

### PR DESCRIPTION
**Pull request by hregis**

### CODE REVIEW

File: htdocs/core/modules/import/import_csv.modules.php [hunk 1]
```diff
@@ -679,7 +680,7 @@ public function import_insert($arrayrecord, $array_match_file_to_database, $obji
 									}
 
 									// Load content of field@table into cache array
-									if (!is_array($this->cachefieldtable[$cachekey])) { // If content of field@table not already loaded into cache
+									if (isset($this->cachefieldtable[$cachekey]) && !is_array($this->cachefieldtable[$cachekey])) { // If content of field@table not already loaded into cache
```
Comment (eldy) [inline htdocs/core/modules/import/import_csv.modules.php:683]: If cachefieldtable[cachekey]  was unset, test was true + warning before the pr, after it seems wrong + no warning. Is this behaviour change done on purpose ?

Comment (hregis) [reply htdocs/core/modules/import/import_csv.modules.php]: @eldy i send a fix

**Comment by hregis** [comment]
@eldy I resolved mass warnings as I discovered them which are more or less related, would you prefer that I make several PRs?

File: htdocs/ticket/class/ticket.class.php [hunk 2]
```diff
@@ -2289,7 +2290,11 @@ public function copyFilesForTicket($forcetrackid = null)
 		$maxheightmini = 72;
 
 		$formmail = new FormMail($this->db);
-		$formmail->trackid = (is_null($forcetrackid) ? 'tic'.$this->id : '');
+		if (empty($public_area)) {
+			$formmail->trackid = (is_null($forcetrackid) ? 'tic'.$this->id : '');
+		} else {
+			$formmail->trackid = $this->track_id;
```
[the remark this answers is not in the document]
Comment (hregis) [reply htdocs/ticket/class/ticket.class.php:2296]: @eldy it's fixed in v18 ! i close this PR, thanks

**Pull request closed without merging**